### ISSUE TITLE
Not seeing "Install Update Now" button in plugin details popup

### ISSUE THREAD

**Issue by MichelleBlanchette** [YahnisElsts/plugin-update-checker#551]
When a newer version becomes available for a plugin, I am not seeing the "Install Update Now" button which shows for plugins hosted via WordPress.org. It should be in the bottom-right corner of the popup modal that appears when clicking "View X.X.X Details" link in the plugin's table of the updates screen in wp-admin.

I am using `\YahnisElsts\PluginUpdateChecker\v5\PucFactory::buildUpdateChecker()` via Composer which shows `yahnis-elsts/plugin-update-checker v5.3`

_Expected behavior with update installation button in plugin's View Details modal_
![Screenshot 2024-01-05 at 8 48 38 AM](https://github.com/YahnisElsts/plugin-update-checker/assets/33374343/f82d3cb4-99e6-44cd-af5e-40e6f138201d)

_Current behavior without update installation button_
![Screenshot 2024-01-05 at 8 48 47 AM](https://github.com/YahnisElsts/plugin-update-checker/assets/33374343/7311201c-e933-4657-aa58-c4b2bcdda0af)

**Comment by YahnisElsts** [comment]
Verify that your update server returns a download link (i.e. `download_url` isn't missing or empty). A quick skim of WordPress core code suggests that it won't show this button if there is no download link.

**Comment by MichelleBlanchette** [comment]
Yes, I saw the same thing in WordPress Core's source code. However, Debug Bar is showing that I am indeed retrieving a proper `download_url` field. (see attached)

Have you confirmed this "Install Update Now" button _is_ supported to work via PUC v5, then? Really, I just wanted confirmation from your end that the button does appear in your own testing of PUC v5.

![Screenshot 2024-01-07 at 8 27 12 PM](https://github.com/YahnisElsts/plugin-update-checker/assets/33374343/6c2662b1-659f-4b73-a1a3-1d1b9a94404e)

**Comment by YahnisElsts** [comment]
I tested it just now, and it seems to show up fine even in the latest alpha version of WordPress. Here's a couple of screenshots from an example plugin that's using PUC release 5.3:

![firefox_2024-01-08_11-17-20](https://github.com/YahnisElsts/plugin-update-checker/assets/2527434/77dfe695-42d6-4f30-8c19-b49001dad2b5)

![firefox_2024-01-08_11-28-46](https://github.com/YahnisElsts/plugin-update-checker/assets/2527434/296163c4-72c3-4954-81fe-f9d80212c5a9)

(The plugin metadata is old, but the PUC version is new.)

Is there anything unusual going on with how PUC is configured in your case? Maybe a custom slug that doesn't match the slug shown in the Debug Bar? Or modifying the `download_url` with custom filters?

**Comment by MichelleBlanchette** [comment]
Those screenshots are incredibly helpful. Thank you! I don't know of anything unique or special about my implementation, though, but our Debug Bar information looks very different from each other as far as the PUC classes etc.

I'll add some deep logging in WordPress Core and see if I can track down what's causing this. I'm not so much concerned about the button not showing as much as it's an indication that there's unexpected behavior happening.

I'll let you know what I find!

**Comment by Thiararapeter** [comment]
Hello, There is an issue with installed plugin. when i click open view detail and te plugin is installed,  be button should be "latest verson installed" but am getting "install now button" mostly works but after plugin updates, yes the plugin is installed but on view details plugin have "install now"
attached will demonstrate what am trying to say.


[cid:7130d225-aa01-4d51-8989-a6777af75774][cid:cb97783a-dad9-4374-8fa2-12f6e0eb1d2e]


Regards,
Thiarara
________________________________
